### PR TITLE
In-app Message View

### PR DESCRIPTION
**Pull request by NghiaTranUIT**
### 📒 Description
This PR introduce the InApp message view.

### 🕶️ Types of changes
**New feature** (non-breaking change which adds functionality)

### 🤯 List of changes
- [x] Get the InAppMessage library code from https://github.com/toggl-open-source/toggldesktop/pull/3578
- [x] Fix one minor error in In-App message library
- [x] Introduce InAppMessage view 
- [x] Support multiple line, depend on the content
- [x] Animation & shadow

### 👫 Relationships
Closes https://github.com/toggl-open-source/toggldesktop/issues/3571

#### Prepare
- Open the database and set the message_seen to `0`

#### Present the message properly with default message
- Open the app
- Check that the message view is appeared properly (Animation, shadow) 
- Able to close by clicking on the Close btn

Screen Shot 2019-12-04 at 16 07 14

#### Present the message properly with long message
- Do `Prepare`
- Go to https://github.com/toggl-open-source/toggldesktop/blob/03a3619c4d60a5c59ad1d9d04f48bdd659ad5a5a/src/ui/osx/TogglDesktop/test2/AppDelegate.m#L1854
- Change the title or description
- Open the app and make sure the content is layout properly

Screen Shot 2019-12-04 at 15 58 45

#### Action button
- Override the URL in message or at above link
- Make sure it open the link properly when clicking

### CODE REVIEW

File: src/context.cc [hunk 1]
```diff
@@ -1530,8 +1530,8 @@ error Context::fetchMessage() {
                 return error("Error parsing update check response body");
             }
             auto messageID = root["id"].asInt64();
-            auto type = root["type"].asInt64();
-            auto days = root["days"].asInt64();
+            auto type = stoi(root["type"].asString());
```
Comment (NghiaTranUIT) [inline src/context.cc]: Fix an exception when the type is not able to cast to Int64

**Comment by skel35** [comment]
was closed accidentally

**Comment by NghiaTranUIT** [comment]
It's ready for a second review @IndrekV 

## Changelogs
- [x] Make it flexible to resize the window -> the content will be layout too
- [x] Add tooltip and close the banner when clicking on the btn
- [x] There is no more text over the screen -> The content would become multiple-line when the width is small
- [x] Add padding for close btn

 GIF (Too big): https://www.dropbox.com/s/par60duhwm8cnhw/2019-12-05%2017.46.14.gif?dl=0

**Comment by IndrekV** [comment]
All issues are fixed! Great job @NghiaTranUIT

**Pull request merged**